server/src/main/java/org/elasticsearch/ingest/IngestService.java

```diff
@@ -880,6 +881,15 @@ private void innerExecute(
                 handler.accept(null);
             }
         });
+        postExecute(ingestDocument, indexRequest);
```

Comment:
I think `postExecute(...)` should be invoked from callback before invoking the  `handler` (line 881).
If pipeline execution is asynchronous then modification to `ingestDocument` may still be made here.

Reply:
Now pipeline execution is synchronous, but it's more suitable to place as you suggest. And I rename it to `postIngest`

Reply:
In the case a processor implementation goes async then it isn't. This is the case for example with the `enrich` and `inference` processors. These processor invoke a (remote) action.

In any case the method is no invoked from the callback. So I think it is good now 👍